template.go

```diff
@@ -38,59 +42,65 @@ type Engine struct {
 	Extension string
 	// layout variable name that incapsulates the template
 	LayoutName string
-	// determines if the engine parsed all templates
-	Loaded bool
-	// reload on each render
-	ShouldReload bool
-	// debug prints the parsed templates
-	Verbose bool
 	// lock for funcmap and templates
 	Mutex sync.RWMutex
 	// template funcmap
-	Funcmap map[string]interface{}
+	Funcmap map[string]any
+	// debug prints the parsed templates
+	verbose bool
+	// determines if the engine parsed all templates
+	loaded bool
+	// reload on each render
+	shouldReload bool
 }
 
 // AddFunc adds the function to the template's function map.
 // It is legal to overwrite elements of the default actions
-func (e *Engine) AddFunc(name string, fn interface{}) IEngineCore {
+func (e *Engine) AddFunc(name string, fn any) IEngineCore {
 	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
 	e.Funcmap[name] = fn
-	e.Mutex.Unlock()
 	return e
 }
 
 // AddFuncMap adds the functions from a map to the template's function map.
 // It is legal to overwrite elements of the default actions
-func (e *Engine) AddFuncMap(m map[string]interface{}) IEngineCore {
+func (e *Engine) AddFuncMap(m map[string]any) IEngineCore {
 	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
 	for name, fn := range m {
 		e.Funcmap[name] = fn
 	}
-	e.Mutex.Unlock()
 	return e
 }
 
 // Debug will print the parsed templates when Load is triggered.
 func (e *Engine) Debug(enabled bool) IEngineCore {
-	e.Verbose = enabled
+	e.verbose = enabled
 	return e
 }
 
 // Delims sets the action delimiters to the specified strings, to be used in
 // templates. An empty delimiter stands for the
 // corresponding default: "{{" and "}}".
 func (e *Engine) Delims(left, right string) IEngineCore {
+	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
 	e.Left, e.Right = left, right
 	return e
 }
 
 // FuncMap returns the template's function map.
-func (e *Engine) FuncMap() map[string]interface{} {
+func (e *Engine) FuncMap() map[string]any {
+	e.Mutex.RLock()
```

Comment:
Yeah, that's the purpose. When I started playing with the `RunParallel()` benchmarks it exposed a lot of concurrency issues since it runs the benchmark using concurrent goroutines